context-config-store/api/golang/context_config.pb.go

```diff
@@ -0,0 +1,610 @@
+// Code generated by protoc-gen-go. DO NOT EDIT.
+// versions:
+// 	protoc-gen-go v1.28.1
+// 	protoc        v3.21.12
+// source: context_config.proto
+
+package golang
```

Comment:
we are using PROTO here because we want to send it over the wire?

Reply:
Hmm, I don't think we will send this over the wire, at least not now. We could have used raw go types, but then it's a _real_ pain to deal with union types, proto has that built-in. We could have used another type generation language as well, but I don't see a compelling argument for bringing another tooling in our repo where proto meets the requirements IMO. Also, this is not user facing at all. Let me know what you think

Reply:
I don't see the the value as keeping something as proto and then storing it in JSON; maybe the marshal/unmarshal story is less complex with a normal Go struct but you don't get unions with it I guess.

Can we stick a comment on top saying that this isn't sent over the wire and the only reason its Proto is that we want to suppport unions?

Reply:
How do you handle union types with go JSON serialization?